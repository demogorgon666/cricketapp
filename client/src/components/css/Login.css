@import url("https://fonts.googleapis.com/css?family=Montserrat:400,500,600&display=swap");

body {
  font-family: "Montserrat", sans-serif;
}

fieldset p {
  margin: 0px;
}

fieldset h1 {
  text-align: center;
}

#msform {
  width: 30vw;
  margin: 100px auto 40px auto;
  /* text-align: center; */
}

#msform fieldset {
  background: white;
  border: 0.5px solid #707070;
  border-radius: 8px;
  padding: 50px 50px;
}

input,
textarea {
  padding: 15px;
  border: 0.5px solid #707070;
  border-radius: 8px;
  margin-bottom: 16px;
  width: 100%;
  box-sizing: border-box;
  font-family: "Montserrat", sans-serif;
  color: #707070;
  font-size: 13px;
}

input:focus,
textarea:focus {
  outline: none;
  border-color: #2980b9;
  color: #2980b9;
  font-weight: 500;
}

::placeholder {
  color: #707070;
}
::placeholder:focus {
  color: #2980b9;
}
span {
  text-decoration-color: #2980b9;
}
.link {
  text-decoration-color: #2980b9;
}
p {
  color: #2980b9;
}

button {
  width: 100%;
  background: #2980b9;
  font-weight: 500;
  color: white;
  box-sizing: border-box;
  border-radius: 8px;
  cursor: pointer;
  padding: 15px;
  border: 0px;
  transition: 0.3s;
  font-family: "Montserrat", sans-serif;
  letter-spacing: 0.6px;
  font-size: 14px;
  margin: 16px 0px;
}

button:hover {
  box-shadow: 0px 4px 6px 0px #c2c2c2;
}

h1 {
  font-size: 30px;
  text-transform: uppercase;
  color: #2980b9;
  margin-bottom: 30px;
}

.errorMessage {
  font-size: 12px;
  margin-bottom: 10px;
  font-weight: 600;
}

@media (max-width: 1200px) {
  #msform {
    width: 35vw;
  }
}

@media (max-width: 992px) {
  #msform {
    width: 40vw;
  }
}

@media (max-width: 767px) {
  #msform {
    width: 50vw;
  }
}

@media (max-width: 575px) {
  #msform {
    width: 70vw;
  }
}

@media (max-width: 425px) {
  #msform {
    width: 90vw;
  }
}
