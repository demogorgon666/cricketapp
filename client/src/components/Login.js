import React, { Component } from "react";

import { login, getUsers } from "../actions/users";
import { connect } from "react-redux";
import Navbar from "./common/Navbar";

import { Link } from "react-router-dom";
import "./css/Login.css";

export class Login extends Component {
	// constructor(props) {
	//   super(props);
	//   this.onLogin = this.onLogin.bind(this);
	// }

	state = {
		user_email: "",
		user_password: "",
		showError: false,
		errorMessage: ""
	};

	componentDidMount() {
		//   axios.get(`http://localhost:5000/api/ecommerce/user/all`).then(res => {
		//     const users = res.data;
		//     this.setState({ users });
		//     console.log(users);
		//   });
		// }
		if (localStorage.getItem("token")) {
			this.props.history.push("/");
		}
	}

	OnChange = event => {
		this.setState({ [event.target.name]: event.target.value });
	};

	onLogin = e => {
		e.preventDefault();
		if (!this.state.user_email || !this.state.user_password) {
			this.setState({
				showError: true,
				errorMessage: "Enter all the fields"
			});
		} else {
			this.setState({
				showError: false,
				errorMessage: ""
			});
			let user = {
				user_email: this.state.user_email,
				user_password: this.state.user_password
			};
			this.props.login(user, this.props.history);
			this.setState({
				user_email: "",
				user_password: ""
			});
		}
	};

	render() {
		return (
			<div>
				<Navbar />
				<form id="msform">
					<fieldset>
						<h1>Login</h1>
						<input
							type="text"
							name="user_email"
							placeholder="Enter Email"
							onChange={this.OnChange}
							value={this.state.user_email}
						/>
						<input
							type="password"
							name="user_password"
							placeholder="Enter Password"
							onChange={this.OnChange}
							value={this.state.user_password}
						/>
						<span
							className="errorMessage"
							style={{
								color: "#c0392b",
								display: this.state.showError ? "block" : "none"
							}}
						>
							{this.state.errorMessage}
						</span>
						<p>
							<Link className="link" to="/resetPassword">
								<span
									style={{
										color: "#2980b9",
										fontWeight: "bold"
									}}
								>
									Forgot Password ?
								</span>
							</Link>
						</p>

						<button onChange={this.onChange} onClick={this.onLogin}>
							Login
						</button>
						<p>
							Not a User ?{" "}
							<Link className="link" to="/register">
								<span
									style={{
										color: "#2980b9",
										fontWeight: "bold"
									}}
								>
									Register Now
								</span>
							</Link>
						</p>
					</fieldset>
				</form>
			</div>
		);
	}
}

const mapStateToProps = state => ({
	users: state.userReducer.users
});

export default connect(
	mapStateToProps,
	{ login, getUsers }
)(Login);
