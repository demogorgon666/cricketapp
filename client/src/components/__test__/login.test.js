import React from "react";
import { shallow, mount } from "enzyme";
import { Login } from "../Login";

const login = jest.fn();
const onLogin = jest.fn();
const wrapper = shallow(<Login login={login} onLogin={onLogin} />);

describe("test  Component", () => {
	it("render the component", () => {
		expect(wrapper).toMatchSnapshot();
	});
	it("there should be a div", () => {
		expect(wrapper.find("div").length).toBe(1);
	});
	it("there should be a p tag", () => {
		expect(wrapper.find("p").length).toBe(2);
	});
	it("there should be a link tag", () => {
		expect(wrapper.find("Link").length).toBe(2);
	});
	it("there should be a h1 tag", () => {
		expect(wrapper.find("h1").length).toBe(1);
	});
	it("there should be a navber", () => {
		expect(wrapper.find("NavBar").length).toBe(0);
	});
	it("there should be a input", () => {
		expect(wrapper.find("input").length).toBe(2);
	});
	it("there should be a form", () => {
		expect(wrapper.find("form").length).toBe(1);
	});
	it("there should be a fieldset", () => {
		expect(wrapper.find("fieldset").length).toBe(1);
	});
	it("there should be a button", () => {
		expect(wrapper.find("button").text()).toBe("Login");
	});
	// it("Test click event", () => {
	//   const tree = shallow(
	//     <Button name='button test' handleClick={mockFn} />
	//   );
	//   expect(onLogin).toHaveBeenCalled(0);
	// });
	// it("should call mock function when button is clicked", () => {
	//   const tree = shallow(<onLogin name="button test" handleClick={onLogin} />);
	//   tree.simulate("click");
	//   expect(onLogin).toHaveBeenCalled();
	// });
});
