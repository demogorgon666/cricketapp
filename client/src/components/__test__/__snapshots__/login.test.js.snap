// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test  Component render the component 1`] = `
<div>
  <Navbar />
  <form
    id="msform"
  >
    <fieldset>
      <h1>
        Login
      </h1>
      <input
        name="user_email"
        onChange={[Function]}
        placeholder="Enter Email"
        type="text"
        value=""
      />
      <input
        name="user_password"
        onChange={[Function]}
        placeholder="Enter Password"
        type="password"
        value=""
      />
      <span
        className="errorMessage"
        style={
          Object {
            "color": "#c0392b",
            "display": "none",
          }
        }
      />
      <p>
        <Link
          className="link"
          to="/resetPassword"
        >
          <span
            style={
              Object {
                "color": "#2980b9",
                "fontWeight": "bold",
              }
            }
          >
            Forgot Password ?
          </span>
        </Link>
      </p>
      <button
        onClick={[Function]}
      >
        Login
      </button>
      <p>
        Not a User ?
         
        <Link
          className="link"
          to="/register"
        >
          <span
            style={
              Object {
                "color": "#2980b9",
                "fontWeight": "bold",
              }
            }
          >
            Register Now
          </span>
        </Link>
      </p>
    </fieldset>
  </form>
</div>
`;
